mujoco_warp/_src/solver_test.py

```diff
@@ -68,6 +69,69 @@ def cost(qacc):
 
       _assert_eq(mjwarp_cost, mj_cost, name="cost")
 
+  def test_init_linesearch(self):
+    """Test linesearch initialization."""
+    mjm, mjd, m, d = test_util.fixture(
+      "constraints.xml",
+      cone=ConeType.PYRAMIDAL,
+      iterations=0,
+      ls_iterations=0,
+    )
+
+    # One step to obtain more non-zeros results
+    mjwarp.step(m, d)
+
+    # Calculate target values
+    def calc_jv(nefc, efc_J, efc_search):
+      jv = np.zeros(nefc)
+      for i in range(nefc):
+        jv[i] += np.sum(efc_J[i, :] * efc_search[:])
+      return jv
+
+    def calc_quad_gauss(efc_gauss, efc_search, efc_Ma, qfrc_smooth, efc_mv):
+      quad_gauss = np.zeros(3)
+      quad_gauss[0] = efc_gauss[0]
+      quad_gauss[1] = np.sum(efc_search[:] * (efc_Ma[0, :] - qfrc_smooth[0, :]))
+      quad_gauss[2] = 0.5 * np.sum(efc_search[:] * efc_mv[:])
+      return quad_gauss
+
+    def calc_quad(nefc, efc_jaref, efc_jv, efc_D):
+      quad = np.zeros((nefc, 3))
+      for i in range(nefc):
+        quad[i, 0] = 0.5 * efc_jaref[i] * efc_jaref[i] * efc_D[i]
+        quad[i, 1] = efc_jv[i] * efc_jaref[i] * efc_D[i]
+        quad[i, 2] = 0.5 * efc_jv[i] * efc_jv[i] * efc_D[i]
+      return quad
+
+    efc_search_np = d.efc.search.numpy()[0]
+    efc_J_np = d.efc.J.numpy()[0]
+    efc_gauss_np = d.efc.gauss.numpy()
+    efc_Ma_np = d.efc.Ma.numpy()
+    efc_Jaref_np = d.efc.Jaref.numpy()[0]
+    efc_D_np = d.efc.D.numpy()[0]
+    qfrc_smooth_np = d.qfrc_smooth.numpy()
+    nefc = d.nefc.numpy()[0]
+
+    target_mv = np.zeros(mjm.nv)
+    mujoco.mj_mulM(mjm, mjd, target_mv, efc_search_np)
+    target_jv = calc_jv(nefc, efc_J_np, efc_search_np)
+    target_quad_gauss = calc_quad_gauss(efc_gauss_np, efc_search_np, efc_Ma_np, qfrc_smooth_np, target_mv)
+    target_quad = calc_quad(nefc, efc_Jaref_np, target_jv, efc_D_np)
+
+    # launch linesearch with 0 iteration just doing the initialization step
+    d.efc.jv.zero_()
+    d.efc.quad.zero_()
+    solver._linesearch(m, d)
+
+    efc_mv = d.efc.mv.numpy()[0]
+    efc_jv = d.efc.jv.numpy()[0]
+    efc_quad_gauss = d.efc.quad_gauss.numpy()[0]
+    efc_quad = d.efc.quad.numpy()[0]
+    _assert_eq(target_mv, efc_mv, name="efc.mv")
```

Comment:
- for these asserts we should swap the order: mujoco warp result first, mujoco (expected) second
- instead of `name="...`, we can just have the field name